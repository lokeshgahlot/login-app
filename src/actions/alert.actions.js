export const ALERT_CLEAR = "ALERT_CLEAR";
export const ALERT_SUCCESS = "ALERT_SUCCESS";
export const ALERT_FAILURE = "ALERT_FAILURE";

export const clear = _ => ({ type: ALERT_CLEAR });
export const success = payload => ({ type: ALERT_SUCCESS, payload });
export const failure = payload => ({ type: ALERT_FAILURE, payload });