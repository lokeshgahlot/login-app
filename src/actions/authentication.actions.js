import {authentication as  authenticationService} from "../services/user.service";
import Cookies from "cookies-js";
import  * as alertActions from "./alert.actions";

export const AUTHENTICATION_REQUEST = "AUTHENTICATION_REQUEST";
export const AUTHENTICATION_SUCCESS = "AUTHENTICATION_SUCCESS";
export const AUTHENTICATION_FAILURE = "AUTHENTICATION_FAILURE";
export const COOKIE_KEY = "login-demo-app";

export const request = _ => ({ type: AUTHENTICATION_REQUEST });
export const success = payload => ({ type: AUTHENTICATION_SUCCESS, payload });
export const failure = payload => ({ type: AUTHENTICATION_FAILURE, payload });

export const authentication = payload => dispatch => {
  request();
  return new Promise((resolve, reject) => {
    authenticationService(payload)
    .then(user => { // success
      dispatch(success(user))
      if (user.token) {
        Cookies.set(COOKIE_KEY, JSON.stringify({token: user.token, id: user.id }), { expires: 60 * 60 * 24 }); // Expires in 1 day
      }
      resolve(user);
      return user;
    })
    .catch(error => { // error
      dispatch(failure(error))
      dispatch(alertActions.failure(error))
      reject(error);
      return error;
    });
  })
}