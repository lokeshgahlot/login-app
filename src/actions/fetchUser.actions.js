import {getUser} from "../services/user.service";
import {failure} from "./alert.actions";

export const FETCH_USER_SUCCESS = "FETCH_USER_SUCCESS";
export const success = (payload) => ({type: FETCH_USER_SUCCESS, payload});

export const fetchUser = payload => dispatch => {
  return getUser(payload)
    .then(data => dispatch(success(data)))
    /* eslint-disable */
    .catch(error => dispatch(failure(error)));
    /* eslint-enable */
};