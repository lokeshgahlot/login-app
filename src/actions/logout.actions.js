import {COOKIE_KEY} from "./authentication.actions";
import Cookies from "cookies-js";

export const LOGOUT = "LOGOUT";

export const logout = () => {
  Cookies.expire(COOKIE_KEY);
  return {
    type: LOGOUT
  };
};