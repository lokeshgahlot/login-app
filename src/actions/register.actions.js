import {register as registerService} from "../services/user.service";
import  * as alertActions from "./alert.actions";

export const REGISTER_REQUEST = "REGISTER_REQUEST";
export const REGISTER_SUCCESS = "REGISTER_SUCCESS";
export const REGISTER_FAILURE = "REGISTER_FAILURE";

export const request = _ => ({ type: REGISTER_REQUEST });
export const success = user => ({ type: REGISTER_SUCCESS, user });
export const failure = error => ({ type: REGISTER_FAILURE, error });

export const register = payload => dispatch => {
  request();
  return new Promise((resolve, reject) => {
    registerService(payload)
    .then(user => { //success
      dispatch(success(user))
      resolve(user);
    })
    .catch(error => { // error
      dispatch(failure(error))
      dispatch(alertActions.failure(error));
      reject(error);
    });
  })
}