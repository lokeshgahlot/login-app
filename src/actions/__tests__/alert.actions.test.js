import * as alertActions from "../alert.actions";

describe("alert.actions", () => {
  it ("clear()", () => {
    expect(alertActions.clear()).toMatchSnapshot();
  });

  it ("success()", () => {
    expect(alertActions.success("Alert success!!")).toMatchSnapshot();
  });

  it ("failure()", () => {
    expect(alertActions.failure("Alert failure!!")).toMatchSnapshot();
  });
});