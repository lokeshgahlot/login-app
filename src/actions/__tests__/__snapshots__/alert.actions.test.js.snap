// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert.actions clear() 1`] = `
Object {
  "type": "ALERT_CLEAR",
}
`;

exports[`alert.actions failure() 1`] = `
Object {
  "payload": "Alert failure!!",
  "type": "ALERT_FAILURE",
}
`;

exports[`alert.actions success() 1`] = `
Object {
  "payload": "Alert success!!",
  "type": "ALERT_SUCCESS",
}
`;
